### PR TITLE
risc-v/espressif: Fix bootloader and app potential RAM overlap

### PR DESCRIPTION
**Pull request by gustavonihei**
## Summary

This PR intends to fix a potential RAM overlapping issue between the bootloader and application images, which has been recently addressed on the ESP-IDF project:
https://github.com/espressif/esp-idf/commit/0fb0be381763332007c269a558e5df9b23f69c18

Furthermore, it proposes another quick fix for the ESP32-C3 linker script. The length of the IROM and DROM segments should be related to the extent of the address space, and not be limited to the capacity of the External Flash device.

## Impact

Fix for a potential issue, at the cost of reduced DRAM area for Espressif RISC-V based chips.

## Testing

CI build pass.

### CODE REVIEW

**Review by tmedicci — APPROVED** [review]

File: boards/risc-v/espressif/common/scripts/esp32c3_flat_memory.ld [hunk 1]
```diff
@@ -41,23 +41,13 @@
 
 /* 2nd stage bootloader iram_loader_seg start address */
 
-#define SRAM_DRAM_END       0x403d0000 - I_D_SRAM_OFFSET
+#define SRAM_DRAM_END       0x403ce710 - I_D_SRAM_OFFSET
```
Comment (pkarashchenko) [inline boards/risc-v/espressif/common/scripts/esp32c3_flat_memory.ld:44]: Optional
```suggestion
#define SRAM_DRAM_END       (0x403ce710 - I_D_SRAM_OFFSET)
```

File: boards/risc-v/espressif/common/scripts/esp32c6_flat_memory.ld [hunk 2]
```diff
@@ -31,26 +31,26 @@
...
+#define SRAM_IRAM_START       0x40800000
+#define SRAM_DRAM_START       0x40800000
 
-#define I_D_SRAM_OFFSET     (SRAM_IRAM_START - SRAM_DRAM_START)
+#define I_D_SRAM_OFFSET       (SRAM_IRAM_START - SRAM_DRAM_START)
 
 /* 2nd stage bootloader iram_loader_seg start address */
 
-#define SRAM_DRAM_END       0x40880000 - I_D_SRAM_OFFSET
+#define SRAM_DRAM_END         0x4086e610 - I_D_SRAM_OFFSET
```
Comment (pkarashchenko) [inline boards/risc-v/espressif/common/scripts/esp32c6_flat_memory.ld:41]: Optional
```suggestion
#define SRAM_DRAM_END         (0x4086e610 - I_D_SRAM_OFFSET)
```

File: boards/risc-v/espressif/common/scripts/esp32c6_flat_memory.ld [hunk 3]
```diff
@@ -31,26 +31,26 @@
...
 
-#define SRAM_IRAM_ORG       (SRAM_IRAM_START)
-#define SRAM_DRAM_ORG       (SRAM_DRAM_START)
+#define SRAM_IRAM_ORG         (SRAM_IRAM_START)
+#define SRAM_DRAM_ORG         (SRAM_DRAM_START)
 
-#define I_D_SRAM_SIZE       SRAM_DRAM_END - SRAM_DRAM_ORG
+#define I_D_SRAM_SIZE         SRAM_DRAM_END - SRAM_DRAM_ORG
```
Comment (pkarashchenko) [inline boards/risc-v/espressif/common/scripts/esp32c6_flat_memory.ld:46]: Optional
```suggestion
#define I_D_SRAM_SIZE         (SRAM_DRAM_END - SRAM_DRAM_ORG)
```

File: boards/risc-v/espressif/common/scripts/esp32h2_flat_memory.ld [hunk 4]
```diff
@@ -31,26 +31,26 @@
...
+#define SRAM_IRAM_START       0x40800000
+#define SRAM_DRAM_START       0x40800000
 
-#define I_D_SRAM_OFFSET     (SRAM_IRAM_START - SRAM_DRAM_START)
+#define I_D_SRAM_OFFSET       (SRAM_IRAM_START - SRAM_DRAM_START)
 
 /* 2nd stage bootloader iram_loader_seg start address */
 
-#define SRAM_DRAM_END       0x40850000 - I_D_SRAM_OFFSET
+#define SRAM_DRAM_END         0x4083efd0 - I_D_SRAM_OFFSET
```
Comment (pkarashchenko) [inline boards/risc-v/espressif/common/scripts/esp32h2_flat_memory.ld:41]: Optional
```suggestion
#define SRAM_DRAM_END         (0x4083efd0 - I_D_SRAM_OFFSET)
```

File: boards/risc-v/espressif/common/scripts/esp32h2_flat_memory.ld [hunk 5]
```diff
@@ -31,26 +31,26 @@
...
 
-#define SRAM_IRAM_ORG       (SRAM_IRAM_START)
-#define SRAM_DRAM_ORG       (SRAM_DRAM_START)
+#define SRAM_IRAM_ORG         (SRAM_IRAM_START)
+#define SRAM_DRAM_ORG         (SRAM_DRAM_START)
 
-#define I_D_SRAM_SIZE       SRAM_DRAM_END - SRAM_DRAM_ORG
+#define I_D_SRAM_SIZE         SRAM_DRAM_END - SRAM_DRAM_ORG
```
Comment (pkarashchenko) [inline boards/risc-v/espressif/common/scripts/esp32h2_flat_memory.ld:46]: Optional
```suggestion
#define I_D_SRAM_SIZE         (SRAM_DRAM_END - SRAM_DRAM_ORG)
```

**Review by pkarashchenko — APPROVED** [review]

File: boards/risc-v/esp32c3/common/scripts/flat_memory.ld [hunk 6]
```diff
@@ -43,7 +43,7 @@
 
 /* 2nd stage bootloader iram_loader_seg start address */
 
-#define SRAM_DRAM_END       0x403d0000 - I_D_SRAM_OFFSET
+#define SRAM_DRAM_END       0x403ce710 - I_D_SRAM_OFFSET
```
[the remark this answers is not in the document]
Comment (gustavonihei) [reply boards/risc-v/esp32c3/common/scripts/flat_memory.ld]: Thanks, I've applied this and the other suggestions.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**